Week3-Project/RAVISHARMA022/Login Backend JAVA/src/main/java/com/example/demo/controller/AuthController.java

```diff
@@ -0,0 +1,126 @@
+package com.example.demo.controller;
+import java.sql.Connection;
+import java.sql.DriverManager;
+import java.sql.ResultSet;
+import java.sql.Statement;
+import java.util.ArrayList;
+import java.util.List;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestMethod;
+import org.springframework.web.bind.annotation.RestController;
+import com.example.Repository.EmpRepository;
+import com.example.demo.EmployeeDetail.AuthenticateUser;
+import com.example.demo.EmployeeDetail.EducationDetail;
+import com.example.demo.EmployeeDetail.EmployeeDetail;
+
+@RestController
+public class AuthController {
+
+	//GetApi for Single User Details
+	@GetMapping("/AuthenticateUser/{Id}")
+	public List<AuthController> index3(@PathVariable String Id){
+		Connection con;
+		List<AuthController> Employees = new ArrayList<AuthController>();
+		ResultSet rs = null;
+		try {
+			String dbURL = "jdbc:sqlserver://DESKTOP-4EEOQT3;database=LoginSystem";
```

Comment:
Instead of making a connection with the sql server everytime separately, you can have a method that returns a connection if it is not null, or create a new connection and return. You can then use this method for all your crud operations.